x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleExplainResponse.java

```diff
@@ -350,8 +349,8 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalString(repositoryName);
             out.writeOptionalString(snapshotName);
             out.writeOptionalString(shrinkIndexName);
+            out.writeOptionalLong(indexCreationDate);
```

Comment:
this needs to only happen (even though it's an optional write, it will write `false` if there's no value)
```
 if (out.getVersion().onOrAfter(Version.V_8_1_0)) 
```

Reply:
Hm, I do not fully follow, isn't this condition always going to be true? This code will only exist above this version right? So if this code exists to create this response then the version is going to be `8.1.0` or newer. Am I missing something?

Reply:
The code does not exist in `8.0` and a mixed cluster of `8.x` would have issues if `8.1. writes to `8.0`

Reply:
Ah, I see, this is the version on the other side of the stream, I thought it was the version of the source! I will fix it.